src/restrictingGuards.ts

```diff
@@ -13,6 +13,18 @@ export const isDateOrDateString = orGuard(
 	isDate
 );
 
+const UUIDRegex = /[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}/;
```

Comment:
You need `^...$` anchors on this, otherwise this will match any string that _contains_ a UUID